packages/lib/src/__tests__/basic-io.test.ts

```diff
@@ -0,0 +1,178 @@
+import { CLIError } from '@oclif/errors'
+
+import { inputAndOutputItem, outputItem, outputList } from '../basic-io'
+import * as format from '../format'
+import { InputProcessor, UserInputProcessor } from '../input'
+import * as inputBuilder from '../input-builder'
+import { IOFormat } from '../io-util'
+import * as output from '../output'
+import * as outputBuilder from '../output-builder'
+import { SmartThingsCommandInterface } from '../smartthings-command'
+import { buildMockCommand } from './test-lib/mock-command'
+import { SimpleType } from './test-lib/simple-type'
+
+
+describe('basic-io', () => {
+	const item = { 'str': 'string', num: 5 }
+	const list = [item]
+	const baseCommand = {
+		...buildMockCommand(),
+		flags: {
+			output: 'output.yaml',
+		},
+	}
+	const formatAndWriteItemSpy = jest.spyOn(format, 'formatAndWriteItem')
+	const formatAndWriteListSpy = jest.spyOn(format, 'formatAndWriteList')
+
+	beforeEach(() => {
+		formatAndWriteItemSpy.mockImplementation(async () => { /* empty */ })
+		formatAndWriteListSpy.mockImplementation(async () => { /* empty */ })
```

Comment:
Since the tests below do not override with any other implementations, these would also be safe to move up to the point of initialization.

Reply:
Sorry, I did not mean to ignore this. I'm gonna merge anyway and fix in another pull request.